docs/src/content/en/docs/workflows/using-with-agents-and-tools.mdx

```diff
@@ -5,346 +5,119 @@ description: "Steps in Mastra workflows provide a structured way to manage opera
-  }),
-  outputSchema: z.object({
-    output: z.string()
-  }),
-
-  execute: async ({ inputData }) => {
-    const { input } = inputData;
+> See [Input Data Mapping](./input-data-mapping.mdx) for more information.
 
-    const  prompt = `Provide facts about the city: ${input}`
-
-    const { text } = await testAgent.generate([
-      { role: "user", content: prompt }
-    ]);
-
-    return {
-      output: text
-    };
-  }
-});
 
-export const testWorkflow = createWorkflow({...})
-  .then(step1)
-  .commit();
-```
+### Calling tools from steps
 
-## Using tools in workflows
+Call tools inside a step’s `execute` function using `.execute()`. This lets you process input and return a result that will be passed to the next step.
 
-To use a tool within a workflow, define it in the usual way, then either add it directly to the workflow using `createStep(testTool)` or, invoke it from within a step's `execute` function using `.execute()`.
-
-### Example tool
-
-The example below uses the Open Meteo API to retrieve geolocation details for a city, returning its name, country, and timezone.
-
-```typescript filename="src/mastra/tools/test-tool.ts" showLineNumbers copy
-import { createTool } from "@mastra/core";
-import { z } from "zod";
+```typescript {8-13,16} filename="src/mastra/workflows/test-workflow.ts" showLineNumbers copy
+import { testTool } from "../tools/test-tool";
 
-export const testTool = createTool({
-  id: "test-tool",
-  description: "Gets country for a city",
-  inputSchema: z.object({
-    input: z.string()
-  }),
-  outputSchema: z.object({
-    country_name: z.string()
-  }),
-  execute: async ({ context }) => {
-    const { input } = context;
-    const geocodingResponse = await fetch(`https://geocoding-api.open-meteo.com/v1/search?name=${input}`);
-    const geocodingData = await geocodingResponse.json();
+const step2 = createStep({
+  // ...
+  execute: async ({ inputData, runtimeContext }) => {
+    const { message } = inputData;
 
-    const { country } = geocodingData.results[0];
+    const response = await testTool.execute({
+      context: {
+        formatted: message
+      },
+      runtimeContext
+    });
 
     return {
-      country_name: country
+      formatted: response.formatted
     };
   }
 });
 ```
 
-### Adding a tool as a step
+> See [Calling Tools](../../examples/tools/calling-tools.mdx) for more information.
 
-In this example, `step1` uses the `testTool`, which performs a geocoding lookup using the provided `city` and returns the resolved `country`.
+### Using tools as steps
 
-The step is added to the workflow using the sequential `.then()` method, allowing it to receive input from the workflow and return structured output. The workflow is finalized with `.commit()`.
+When using a tool as a step, it must be wrapped with `createStep()`. The tool's input must match the shape expected by its schema. Use `.map()` to transform the workflow input or previous step’s output as needed before invoking the tool.
```

Comment:
Nit, but using "must" back to back doesn't read as well as it can

I'm also not sure about the language "When using a tool as a step, it must be wrapped with `createStep()`" 

"Must" sounds a tad forceful - could sound more collaborative and supportive (nit, but could be impactful if we apply this tone throughout  our writing)